tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.cpp

```diff
@@ -187,16 +188,31 @@ void EltwiseBinary::validate(const std::vector<Tensor>& input_tensors) const {
             TT_FATAL(this->output_mem_config.memory_layout == TensorMemoryLayout::INTERLEAVED);
         }
     }
+    if(output_tensors.empty() || !output_tensors.at(0).has_value()){
+        // If the user decided to not use any optional output tensors, then this would be empty or would be a nullptr.
+        return;
+    }
+    const auto output_shape_required = this->compute_output_shapes(input_tensors);
```

Comment:
After discussing with @arakhmati and @eyonland, if output tensor is provided, we should validate with only the output tensor, and ignore the output dtype and output memcfg args, so we should shuffle where these assertions are currently taking place

Reply:
moved this assertion for output_tensor to take place right after input tensors validation